test/subtype.jl

```diff
@@ -796,6 +800,11 @@ function test_intersection()
                    (@UnionAll X<:(@UnionAll T Tuple{T,T}) Ref{X}))
     @testintersect(Ref{@UnionAll T @UnionAll S Tuple{T,S}},
                    Ref{@UnionAll T Tuple{T,T}}, Bottom)
+
+    @testintersect(Tuple{T,T} where T<:Union{UpperTriangular, UnitUpperTriangular},
+                   Tuple{AbstractArray{T,N}, AbstractArray{T,N}} where N where T,
```

Comment:
I would support `where T, S` meaning `where S where T`:

1. It matches the comma-separated order inside of `Array{T<:Number, S<:Vector{T}}`;
2. If the other order is more intuitive, you can just use nested `where`s.